Modules/xSharePoint/DSCResources/MSFT_xSPFeature/MSFT_xSPFeature.psm1

```diff
@@ -47,23 +49,55 @@ function Set-TargetResource
         [parameter(Mandatory = $true)]  [System.String] $Url,
         [parameter(Mandatory = $true)]  [ValidateSet("Farm","WebApplication","Site","Web")] [System.String] $FeatureScope,
         [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount,
-        [parameter(Mandatory = $true)]  [ValidateSet("Present","Absent")] [System.String] $Ensure
+        [parameter(Mandatory = $true)]  [ValidateSet("Present","Absent")] [System.String] $Ensure,
+        [parameter(Mandatory = $false)] [System.String] $Version
     )
 
     $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
         $params = $args[0]
         
 
-        $runParams = @{}
-        $runParams.Add("Identity", $params.Name)
+        $runParams = @{ 
+            Identity = $params.Name 
+            Verbose = $false
+        }
+
         if ($params.FeatureScope -ne "Farm") {
             $runParams.Add("Url", $params.Url)
         }
 
+        # Parameters for get
+        $checkParams = @{ 
+            Identity = $params.Name 
+            Verbose = $false    
+        }
+        
+        if ($params.FeatureScope -eq "Farm") 
+        {
+            $checkParams.Add($params.FeatureScope, $true)
+        } 
+        else
+        {
+            $checkParams.Add($params.FeatureScope, $params.Url)
+        }
```

Comment:
This results in another roundtrip to the Server and loading SharePoint snapin again. I took it out for performance reasons - but I understand your concern for maintainability and moved it back in.
But I wouldn’t check the version here. If we’re at the point, that we want to enable a feature at a given scope and it is already activated we have to deactivate it first anyway. Ok with that? I passed in all current values anyway to easily add more logic to it.